src/main/java/com/adobe/aem/compgenerator/utils/HTMLUtils.java

```diff
@@ -0,0 +1,103 @@
+package com.adobe.aem.compgenerator.utils;
+
+import com.adobe.aem.compgenerator.Constants;
+import com.adobe.aem.compgenerator.models.GenerationConfig;
+import com.adobe.aem.compgenerator.models.Property;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class HTMLUtils {
+
+    public static String renderHtml(GenerationConfig generationConfig) {
```

Comment:
Rendered HTML ends up a little bit sloppy with first line indented (because the template file has the indent spacing in it) but the remainder of lines not indented.

```
<sly data-sly-use.demoCompModel="com.newco.aem.base.core.models.DemoComp"
     data-sly-test.configured="${false}">
    <p>Global Textfield Test: ${demoCompModel.globalTextfieldTest}</p>
<p>Shared Textfield Test: ${demoCompModel.sharedTextfieldTest}</p>
<p>Textfield Test: ${demoCompModel.textfieldTest}</p>
<p>label_not_defined: ${demoCompModel.checkTest ? 'checked' : 'unchecked'}</p>
<p>Pathfield Test: ${demoCompModel.pathfieldTest}</p>
<p>Textarea Test: ${demoCompModel.textareaTest}</p>
<p>Select Date: ${demoCompModel.dateTest.time.toGMTString}</p>
<p>Select Test: ${demoCompModel.selectTest}</p>
<p>label_not_defined: ${demoCompModel.radioTest}</p>
<p>label_not_defined: ${demoCompModel.hiddenTest}</p>
<p>Numberfield Test: ${demoCompModel.numberfieldTest}</p>
<p>Image Test: <img src="demoCompModel.imageTest.src}"/></p>
<div data-sly-list="${demoCompModel.colors}">
	<p>Colors: ${item}</p>
</div>
<div data-sly-list="${demoCompModel.links}">
	<p>Path: ${item.path}
	Label: ${item.label}
	</p>
</div>

</sly>
```

Extra credit if you can fix the indentation to keep things nice and neat :)